configuration/doc/ONELINECRBOT.md

```diff
@@ -0,0 +1,108 @@
+
+## Github Workflows
+
+Github workflows are activated by placing a YAML file in the .github/workflows
+directory. By placing the below file into that directory, i.e. storing the full
+below content in the file .github/workflows/one-line-cr-bot.yml any future pull
+request will receive code analysis.
+
+The file has sections labeled "[ACTION REQUIRED]" which likely need to be
+adapted to meet the configuration of your project.
+
+In case the analysis creates many false positive alarms for your project,
+consider to pass the parameter '-y' to the one-line-cr-bot.sh call. This will
+still run the analysis, but will not fail in case of errors. When doing so,
+spotting new defects becomes a more manual process, but will not block your pull
+requests unnecessarily.
+
+    # Author: Norbert Manthey <nmanthey@amazon.de>
+    #
+    # This workflow will present introduced defects of a pull request to a given
+    # branch of a package.
+    #
+    # The workflow has locations labeled '[ACTION REQUIRED]' where adaptation for
+    # your build might be required, as well as where to compare the findings to.
+    #
+    # To learn more about the available options, check the CLI parameters of the
+    # script 'one-line-cr-bot.sh' in https://github.com/awslabs/one-line-scan.git
+    name: One Line CR Bot
+
+    on:
+    pull_request:
+        # [ACTION REQUIRED] Set the branch you want to analyze PRs for
+        branches: [ mainline ]
+
+    # [ACTION REQUIRED] Use this, if you want analysis for push to repository as well
+    push:
+        branches: [ mainline ]
+
+    jobs:
+    build:
+
+        runs-on: ubuntu-latest
+
+        # Get the code, fetch the full history to make sure we have the compare commit as well
+        steps:
+        - uses: actions/checkout@v2
+        with:
+            fetch-depth: 0
+
+        # one-line-cr-bot.sh will get infer and cppcheck, if not available
+        - name: Install CppCheck Package
+        env:
+            # This is needed in addition to -yq to prevent apt-get from asking for user input
+            DEBIAN_FRONTEND: noninteractive
+        # [ACTION REQUIRED] Add your build dependencies here, drop cppcheck to get latest cppcheck
+        run: |
+            sudo apt-get install -y cppcheck
+
+        # Get the compare remote
+        - name: Setup Compare Remote
+        # [ACTION REQUIRED] Add the https URL of your repository
+        run: git remote add compare https://github.com/awslabs/ktf.git
+        - name: Fetch Compare Remote
+        run: git fetch compare
+
+        # Get one-line-scan, the tool we will use for analysis
+        - name: Get OneLineScan
+        run:  git clone -b one-line-cr-bot https://github.com/awslabs/one-line-scan.git ../one-line-scan
+
+        # Check how repository is setup
+        - name: Be Verbose about Git Setup
+        run: |
+            git remote -v
+            git branch -a
+            git log --pretty=oneline --decorate --graph | head -n 10
```

Comment:
How about `... --graph HEAD~10..HEAD` instead of using a shell pipe?

Reply:
For repositories with a more complicated history, using HEAD~10..HEAD might result in presenting many more commits. Hence, I limit this with the shell pipe.